src/Core/Settings.cpp

```diff
@@ -7037,6 +7037,9 @@ Possible values:
 - positive integer - max length of the name,
 - 0 — always,
 - negative integer - never.
+)", 0) \
+    DECLARE(UInt64, max_subcolumns, 10000, R"(
```

Comment:
Unfortunately, this approach doesn't limit the number of dynamic subcolumns inside `JSON`/`Dynamic` types, because we determine dynamic subcolumns on insert when we have the data. So either we state that this limit doesn't include dynamic subcolumns, or we have to implement the check during insert into a table. We need to discuss what we want